### PR TITLE
Update the CI

### PR DESCRIPTION
**Pull request by thfroitzheim**
I updated the CI comparably to the tblite based fully on pip. I removed also the build and python yamls for micomamba. 

Now the full CI including coverage and wheels works again.

### CODE REVIEW

File: .github/workflows/wheel.yml [hunk 1]
```diff
@@ -1,16 +1,52 @@
...
+on:
+  push:
+  pull_request:
+  workflow_dispatch:
+  release:
+    types: [published]
 
 env:
   # https://github.com/actions/checkout/issues/1809
   ACTIONS_ALLOW_USE_UNSECURE_NODE_VERSION: true
```
Comment (marvinfriede) [inline .github/workflows/wheel.yml:12]:
```suggestion
```

Comment (marvinfriede) [reply .github/workflows/wheel.yml:12]: That was a temporary fix, which is probably not needed anymore.

File: .github/workflows/wheel.yml [hunk 2]
```diff
@@ -19,75 +55,96 @@ jobs:
...
-    container: condaforge/linux-anvil-cos7-x86_64
+    runs-on: ${{ matrix.os }}
     strategy:
       fail-fast: false
       matrix:
-        python: ['3.7', '3.8', '3.9', '3.10', '3.11']
+        os:
+        - ubuntu-latest
+        - macos-latest
+        python: ['39', '310', '311', '312']
```
Comment (awvwgk) [inline .github/workflows/wheel.yml:71]: Can we include '313' here?

File: .github/workflows/wheel.yml [hunk 3]
```diff
@@ -19,75 +55,96 @@ jobs:
...
-    runs-on: ubuntu-latest
-    container: condaforge/linux-anvil-cos7-x86_64
+    runs-on: ${{ matrix.os }}
     strategy:
       fail-fast: false
       matrix:
-        python: ['3.7', '3.8', '3.9', '3.10', '3.11']
+        os:
+        - ubuntu-latest
+        - macos-latest
```
[the remark this answers is not in the document]
Comment (marvinfriede) [reply .github/workflows/wheel.yml:70]: `macos-12`  runners are deprecated.

**Review by marvinfriede — APPROVED** [review]

File: .github/workflows/wheel.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (awvwgk) [reply .github/workflows/wheel.yml:70]: In this case let's use `macos-13` which should still support x86_64 builds.

[the remark this answers is not in the document]
Comment (marvinfriede) [reply .github/workflows/wheel.yml:70]: Agreed. Just for completeness:
- `macos-12` runner image will be removed on December 3rd
- `macos-13`: last x86_64-based runner
- starting with `macos-14`, runners are ARM-based
- `macos-15` is `macos-latest`

[the remark this answers is not in the document]
Comment (thfroitzheim) [reply .github/workflows/wheel.yml:70]: Ok, I will include specific macos-13 tests with x86_64 again.

[the remark this answers is not in the document]
Comment (thfroitzheim) [reply .github/workflows/wheel.yml:70]: I added macos-13 back to both the wheels and the build.

**Pull request merged**